src/apps/the-exchange/components/ExchangeAction/ExchangeAction.tsx

```diff
@@ -1,46 +1,105 @@
+import { useContext, useState } from 'react';
+import { ethers } from 'ethers';
+import { hasThreeZerosAfterDecimal } from '../../utils/converters';
+
+// hooks
+import useGlobalTransactionsBatch from '../../../../hooks/useGlobalTransactionsBatch';
+import useBottomMenuModal from '../../../../hooks/useBottomMenuModal';
+import { useEtherspotSwaps } from '@etherspot/transaction-kit';
+
+// context
+import { SwapDataContext } from '../../context/SwapDataProvider';
+
+// components
 import TokenLogo from '../TokenLogo/TokenLogo';
 import Body from '../Typography/Body';
 import NumberText from '../Typography/NumberText';
+import BodySmall from '../Typography/BodySmall';
+import { CircularProgress } from '@mui/material';
+
+// images
 import ArrowRight from '../../images/arrow-right.png';
-import { useContext } from 'react';
-import { SwapDataContext } from '../../context/SwapDataProvider';
-import useGlobalTransactionsBatch from '../../../../hooks/useGlobalTransactionsBatch';
 
 const ExchangeAction = () => {
-    const { bestOffer, swapToken } = useContext(SwapDataContext);
+    const { bestOffer, swapToken, receiveToken } = useContext(SwapDataContext);
+    const [errorMessage, setErrorMessage] = useState<string>('');
+    const [ isAddingToBatch, setIsAddingToBatch] = useState<boolean>(false);
     const { addToBatch } = useGlobalTransactionsBatch();
+    const { showSend } = useBottomMenuModal();
+    const { prepareCrossChainOfferTransactions } = useEtherspotSwaps();
 
-    const onClickToExchange = () => {
-        if (bestOffer) {
-            addToBatch({
-                title: 'Swap assets',
-                chainId: swapToken?.chainId || 0,
-                to: bestOffer.transactions[0].to,
-                value: bestOffer.transactions[0].value,
-                data: bestOffer.transactions[0].data,
-              });
+    const onClickToExchange = async () => {
+        setErrorMessage('');
+
+        if (!bestOffer) {
+            setErrorMessage('No offer available to exchange');
```

Comment:
Please remember to help the user:

```suggestion
            setErrorMessage('No offer was found! Please try changing the amounts to try again');
```

Reply:
You are right, done!